File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;

public class Main {
    private static int N, M, V;
    private static boolean[] visited;
    private static int [][]arr;
    private static Queue<Integer> q;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        N = sc.nextInt();
        M = sc.nextInt();
        V = sc.nextInt();

        arr = new int[N][N];
        visited = new boolean[N];
        q = new LinkedList<>();

        for (int i = 0; i < M; i++) {
            int y = sc.nextInt()-1;
            int x = sc.nextInt()-1;
            arr[y][x] = 1;
            arr[x][y] = 1;
        }

        visited[V - 1] = true;
        dfs(V - 1);

        System.out.println();

        visited = new boolean[N];
        visited[V - 1] = true;
        q.add(V - 1);
        bfs(V - 1);

        while (!q.isEmpty()) {
            Integer v = q.poll();
            System.out.printf(v+1 + " ");
            bfs(v);
        }

    }

    private static void dfs(int v) {
        System.out.print(v+1 + " ");
        for (int i = 0; i < N; i++) {
            if (arr[v][i] == 1 && !visited[i]) {
                visited[i] = true;
                dfs(i);
            }
        }
    }

    private static void bfs(int v) {
        for (int i = 0; i < N; i++) {
            if (arr[v][i] == 1 && !visited[i]) {
                visited[i] = true;
                q.add(i);
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from copy import deepcopy
from collections import deque
n, m, v = map(int, input().split())

arr = [[0 for _ in range(n+1)] for _ in range(n+1)]
arr2 = deepcopy(arr)

for _ in range(m):
    i,j = map(int,input().split())
    arr[i][j] = 1
    arr[j][i] = 1
    arr2[i][j] = 1
    arr2[j][i] = 1

def dfs(start):
    print(start, end=' ')
    for i in range(1,n+1):
        if arr[start][i] == 1:
            for k in range(1,n+1):
                arr[k][i] = 0
            dfs(i)

queue = deque()
def bfs():
    global queue
    while queue:
        start = queue.popleft()
        print(start, end=' ')
        for i in range(1, n + 1):
            if arr2[start][i] == 1:
                for k in range(1, n + 1):
                    arr2[k][i] = 0
                queue.append(i)



for i in range(1,n+1):
    arr[i][v] = 0
dfs(v)
print()
queue.append(v)
for i in range(1,n+1):
    arr2[i][v] = 0
bfs()